core/sys/linux/sys.odin

```diff
@@ -216,7 +216,17 @@ rt_sigprocmask :: proc "contextless" (mask_kind: Sig_Mask_Kind, new_set: ^Sig_Se
 	return Errno(-ret)
 }
 
-// TODO(flysand): ioctl
+/*
+	Control devices. The ioctl syscall is a bit special because
+	its argument is usually a pointer to some driver-specific structure.
+	The request value is device-specific. Consult your LibC implementation's
+	ioctls.h file to learn more.
+	Available since Linux 1.0.
+*/
+ioctl :: proc "contextless" (fd: Fd, request: u64, arg: u64) -> (Errno) {
```

Comment:
The prototype seems to be wrong. I do not advice using man(2) for figuring out the prototype of syscall functions as man only documents libc, and rarely the underlying kernel implementation.

See [linux source code for ioctl](https://elixir.bootlin.com/linux/latest/source/fs/ioctl.c#L857)  to see the actual prototype for this syscall.

In this case `request` should have type `i32`, and arg should be the size of `uint` or `uintptr`.

Also `u64` and `i64` should probably not appear in parameters for the syscalls, as this won't necessarily apply to 32-bit targets. Syscalls only take parameters in registers, and on 32-bit platforms only have up to 32-bit GP register size.